ext/AdvancedPSLibtaskExt.jl

```diff
@@ -58,30 +81,25 @@ end
 # step to the next observe statement and
 # return the log probability of the transition (or nothing if done)
 function AdvancedPS.advance!(trace::LibtaskTrace, isref::Bool=false)
-    taped_globals = trace.model.ctask.taped_globals
-    rng = taped_globals.rng
+    rng = get_rng(trace)
     isref ? AdvancedPS.load_state!(rng) : AdvancedPS.save_state!(rng)
     AdvancedPS.inc_counter!(rng)
-
-    Libtask.set_taped_globals!(trace.model.ctask, TapedGlobals(rng, taped_globals.other))
-    trace.rng = rng
-
+    set_rng!(trace, rng)
     # Move to next step
     return Libtask.consume(trace.model.ctask)
 end
 
-# create a backward reference in task_local_storage
-function AdvancedPS.addreference!(task::Libtask.TapedTask, trace::LibtaskTrace)
-    rng = task.taped_globals.rng
-    Libtask.set_taped_globals!(task, TapedGlobals(rng, trace))
-    return task
+"""
```

Comment:
It seems that `addreference!` is no longer needed, given that it stores a self-reference? If so, can we remove it?

Reply:
The last bug that I had here was in fact a missing call to `addreference!`, I think in `fork`. We also need to call it in Turing's `particle_mcmc`, to keep the reference in sync so that we can access the varinfo of the trace from within the TapedTask. There's probably a way to get rid of it, but that would require some refactoring, which would require me learning better what is going on. If there's a plan to merge AdvancedPS into Turing proper, do you think that's worth the effort now?